File: content-solutions/stack/warmer-wait.java
import java.util.*;

class Solution {
    public int[] solution(int[] temps) {
        int[] answer = new int[temps.length];
        Deque<Integer> stack = new ArrayDeque<>();
        for (int today = 0; today < temps.length; today++) {
            while (!stack.isEmpty() && temps[stack.peek()] < temps[today]) {
                int day = stack.pop();
                answer[day] = today - day;
            }
            stack.push(today);
        }
        return answer;
    }
}

File: content-solutions/stack/warmer-wait.js
function solution(temps) {
  const answer = new Array(temps.length).fill(0);
  const stack = [];
  temps.forEach((temp, today) => {
    while (stack.length > 0 && temps[stack[stack.length - 1]] < temp) {
      const day = stack.pop();
      answer[day] = today - day;
    }
    stack.push(today);
  });
  return answer;
}
